src/pixelflow/core.py

```diff
@@ -170,23 +173,42 @@ def pixelflow(
 
     # If image is ZYX then use regionprops_3D
     elif dim_labels == "ZYX":
-        # calculate the regionprops features: bbox and centroid
+        features_2d = ("label", "bbox", "centroid")
+        # check whether image_intensity is requested
+        if "image_intensity" in features:
+            # remove image_intensity from features_3d and add to features_2d
+            features_3d = tuple(item for item in features if item != "image_intensity")
+            features_2d += ("image_intensity",)
+        else:
+            features_3d = features
```

Comment:
Could use a `set` for operations like this, e.g.:

```python
features_3d = tuple(set(features).difference({"image_intensity"}))
```

It should work without needing the if statement